pkg/integrations/squid_exporter/squid_exporter.go

```diff
@@ -0,0 +1,111 @@
+package squid_exporter
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	se "github.com/boynux/squid-exporter/collector"
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/prometheus/client_golang/prometheus"
+
+	integrations_v2 "github.com/grafana/agent/pkg/integrations/v2"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+)
+
+// DefaultConfig is the default config for the squid integration
+var DefaultConfig = Config{
+	Address:  "localhost:3128",
+	Username: "",
+	Password: "",
+}
+
+var (
+	errNoAddress  = errors.New("no address was provided")
+	errNoHostname = errors.New("no hostname in provided address")
+	errNoPort     = errors.New("no port in provided address")
+)
+
+// Config is the configuration for the squid integration
+type Config struct {
+	Address  string `yaml:"address"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+	Host     string
+	Port     int
+}
+
+func (c *Config) validate() error {
+	if c.Address == "" {
+		return errNoAddress
+	}
+
+	host, scheme, err := net.SplitHostPort(c.Address)
+	if err != nil {
+		return err
+	}
+
+	if host == "" {
+		return errNoHostname
+	}
+	c.Host = host
+
+	if scheme == "" {
+		return errNoPort
+	}
+
+	port, _ := strconv.Atoi(scheme)
+	c.Port = port
+
+	return nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
```

Comment:
Is the port for squid generally just localhost:3128? I wonder if its worth trying to determine if only a host is provided if we can assume the default port if that's the case.

Reply:
I believe that a default port for squid is 3128, but apparently 8080 is also common, but you're proposing that if a hostname is provided that then we could assume that the port is 3128 and fill it in?

Reply:
I'm not super familiar with Squid but yeah if there is a reasonable default AND the library doesn't assume it and we are validating existence of the port I think it might be applicable to assume a user meant whatever is the default port.